### ISSUE TITLE
Distinction between consecutive tasks

### ISSUE THREAD

**Issue by Gleek** [Fuco1/org-timeline#20]
Hello Matus,

Thank you for writing this package.

Is it possible to add a small gap between consecutive tasks so that distinction can be made among them?

This is done in google calendar by always adding a pixel or two of whitespace in the end
<img width="346" alt="image" src="https://user-images.githubusercontent.com/3756804/109510339-e2f94680-7ac7-11eb-9442-3c74f29d998b.png">

This is not apparent when looking at the org-timeline view
<img width="1066" alt="image" src="https://user-images.githubusercontent.com/3756804/109512733-687df600-7aca-11eb-97e2-11d9b6296044.png">

The 10:30AM-12AM window in this case has two tasks in it, which isn't clear unless we hover on them.

I personally use two colors alternatively in my agenda view to create slight distinction in the agenda log.
<img width="459" alt="image" src="https://user-images.githubusercontent.com/3756804/109511533-256f5300-7ac9-11eb-91fb-c1b47a99a2d7.png">

Incorporating the gap would make it much cleaner, though I'll be happy to have the alternative color hack as well, in the case where implementing the former is very complicated :) 

Thanks.

**Comment by aizensoosuke** [comment]
For now, pull request #19 implements a distinction by adding an overline on every other consecutive task. 
I'm not sure it's possible to add a "small" gap (i.e. less than a whole block), but I might look into it later.
As for colors, that could be an option too. Maybe @Fuco1 has a preference.

This is how it looks with #19 :
![image](https://user-images.githubusercontent.com/26968781/109665058-155d7f00-7b6e-11eb-91c1-88cfda1610f3.png)

Unfortunately putting boxes around blocks will not work, because emacs fuses them even if they are different colors. Putting a zero-width box around every other block almost works, but it still messes with the alignment and makes the timeline unreadable.

**Comment by Gleek** [comment]
Hey @aizensoosuke. Thanks for this
> This is how it looks with #19 :

Interesting. For some reason the overline is hard to see in my case.
 Although it's close to what I want after setting `org-timeline-overlap-in-new-line` and `org-timeline-show-title-in-blocks` to `t`
<img width="156" alt="image" src="https://user-images.githubusercontent.com/3756804/109673876-74d28380-7b9c-11eb-8db8-f20b3fef1530.png">

This is how 3 tasks are set
<img width="333" alt="image" src="https://user-images.githubusercontent.com/3756804/109673139-ba428100-7b9b-11eb-80b6-c7565ba66189.png">


> Unfortunately putting boxes around blocks will not work, because emacs fuses them even if they are different colors, and because it messes with the alignment:

Oh, I was thinking if we could somehow reduce the end time for the duration by a minute while rendering the block then we could get the desired result. This reduction should only happen when rendering that block and not when detecting the overlap.

So for this org file
```org
* TODO Task1
<2021-03-03 Wed 10:00-11:00>
* TODO Task2
<2021-03-03 Wed 11:00-11:30>
```
If we change the end time for Task1 to 10:59 we will get a single character gap
<img width="353" alt="image" src="https://user-images.githubusercontent.com/3756804/109675068-72245e00-7b9d-11eb-916d-844f19c17838.png">

You seem to be better equipped to answer if such an implementation is feasible.

**Comment by aizensoosuke** [comment]
> Oh, I was thinking if we could somehow reduce the end time for the duration by a minute while rendering the block then we could get the desired result. This reduction should only happen when rendering that block and not when detecting the overlap.

This is possible, I'll implement it quickly as an option. The problem with this is that you won't be able to differentiate between blocks that last a whole hour and blocks that are a little than more 40 minutes. That is because each hour is divided into 6 blocks of 10min that can't be divided.

It will also delete very short blocks that would otherwise be rendered, but are directly followed by another block.

**Comment by aizensoosuke** [comment]
I'm letting you know that I implemented it. You'll have to setq `org-timeline-space-out-consecutive` to `t`. See #19

**Comment by aizensoosuke** [comment]
The way we're checking for overlaps makes this break `org-timeline-overlap-in-new-line` this will have to be fixed later. Hopefully this is good enough for you, for now.

**Comment by Gleek** [comment]
Great, it unfortunately didn't work for me
```
* TODO Task1
<2021-03-03 Wed 10:00-11:00>
* TODO Task2
<2021-03-03 Wed 11:00-11:30>
```
<img width="381" alt="image" src="https://user-images.githubusercontent.com/3756804/109683003-f0383300-7ba4-11eb-9fb7-bc52d1c5a990.png">

**Comment by Gleek** [comment]
I just noticed you pushed another commit. It seems to work with that. Thank you for the effort @aizensoosuke 😄. I'm impressed with your speed to implement this.

<img width="388" alt="Screenshot 2021-03-02 at 10 25 29 PM" src="https://user-images.githubusercontent.com/3756804/109684365-34780300-7ba6-11eb-8625-7effe3b8293f.png">

The only issue that now remains for me it seems is that the blocks don't perfectly align with the time slots. Maybe I should create another issue for this.

The `Task1` task, for example in the screenshot above, starts exactly at 10:00AM but it seems as if it starts slightly before that.

**Comment by aizensoosuke** [comment]
Not really, what I did makes all the starting times shift.

> The only issue that now remains for me it seems is that the blocks don't perfectly align with the time slots. Maybe I should create another issue for this.

Yes, I'm fixing this, I should push the fix in a couple minutes.

**Comment by aizensoosuke** [comment]
> I'm impressed with your speed to implement this.

Thank you! I happened to need the exact same thing you needed just today, so I was already working on it anyway. I noticed your issue only after.

> Thank you for the effort @aizensoosuke

Glad to help!

It should be working by now 👌

**Comment by Gleek** [comment]
> It should be working by now 👌

Do you mean the alignment of blocks in timeslots?  That still doesn't seem to work for me, if that's the case.

The gaps work perfectly for me though. 👍

**Comment by aizensoosuke** [comment]
> Do you mean the alignment of blocks in timeslots? That still doesn't seem to work for me, if that's the case.

Did you use the last commit to #19?

If so, could you show me a screeshot?

**Comment by Gleek** [comment]
> If so, could you show me a screeshot?

<img width="572" alt="image" src="https://user-images.githubusercontent.com/3756804/109689001-e1ed1580-7baa-11eb-8894-d3e65dc2a57b.png">

```
  |10:00
  Task1
```
should ideally look
```
  |10:00
  |Task1
```

**Comment by aizensoosuke** [comment]
> The Task1 task, for example in the screenshot above, starts exactly at 10:00AM but it seems as if it starts slightly before that.

Oh sorry, we're talking about separate issues! I read too fast.

I had introduced a bug that shifted start times of blocks with my implementation of space-out-consecutive. That is fixed now.

What you're talking about is, indeed, a separate issue. We could probably change it by using a different character for the vertical separation of hours, but it would be pretty asymmetrical and I'm not sure that would please the eye.

>     |10:00
>      Task1

This was a choice made by @Fuco1. I'll let him decide if he wants to keep it, but I think you're right. I'll implement that later, unless @Fuco1 has an objection.

**Comment by aizensoosuke** [comment]
I realized it was just 2 characters in the code, so here you go. See #19

**Comment by Gleek** [comment]
> I realized it was just 2 characters in the code, so here you go. See #19

Awesome, it works fairly well. Although, the end moves 1 offset forward as well
<img width="96" alt="image" src="https://user-images.githubusercontent.com/3756804/109693317-86715680-7baf-11eb-81ce-d6d3cda6ebad.png">
It seems that the size needs to be shortened by a character to make it fit. I'll try to look into the code this weekend and see if I can make it happen locally. Will create a separate issue otherwise. 

Anyways, thank for all the help.

**Comment by aizensoosuke** [comment]
>  Will create a separate issue otherwise.

This should probably happen at this point.

What you're asking for is that events fit in their slot only if they're one hour long and start at an o'clock time? 
That would mean block length doesn't represent task length accurately anymore. As this is @Fuco1's project, he'll decide, but you should probably just fork this and add:

```elisp
    (when (and (eq (mod beg 60) 0) (eq (mod end 60) 0))
                  (setq end-pos (- end-pos 1))
                  (setq block-length (- block-length 1)))
```
after line 442 of org-timeline.el.

**Comment by aizensoosuke** [comment]
So, to go back to the initial topic of this issue, I think having a separating character might make this clearer than it currently is.
After trying a lot of different unicode characters, I personally find this one to be nice: 
![image](https://user-images.githubusercontent.com/26968781/109800374-5fea0480-7c1d-11eb-8311-74e35939c01e.png)
But there is the issue of losing room for small task's titles.

Maybe just having the overline making consecutive events distinct is the way to go. @Fuco1
